src-tauri/src/main.rs

```diff
@@ -1,22 +1,100 @@
 // Prevents additional console window on Windows in release, DO NOT REMOVE!!
 #![cfg_attr(not(debug_assertions), windows_subsystem = "windows")]
+extern crate unix_named_pipe;
+extern crate ctrlc;
 
 use dotenv::dotenv;
-use std::thread;
+use std::{thread, fs, env};
+use std::path::Path;
 use tpulse::watcher::{AFKSettings, AFKWatcher};
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::thread::JoinHandle;
+use unix_named_pipe::FileFIFOExt;
+use std::io::{self, Read, BufRead};
+
+
+
 
 fn main() {
     dotenv().ok();
     env_logger::init();
 
     let afk_settings = AFKSettings::new(5000, 500);
     let afk_watcher = AFKWatcher::new(&afk_settings);
-
-    let afk_watch = thread::spawn(move || afk_watcher.run());
-
+    // let afk_watch = thread::spawn(move || afk_watcher.run());
+    let open_server_pipe = thread::spawn(move || {
+        let pipe_path = "/home/tan17112003/Desktop/myunixnamedpipe";
+        let running = make_loop_flag();
+        let file = try_open(&pipe_path).expect("could not open pipe for reading");
+        let mut reader = io::BufReader::new(file);
+        while running.load(Ordering::SeqCst) {
+            let mut payload= String::new();
+            let res = reader.read_line(&mut payload);
+            if let Err(err) = res {
+                match err.kind() {
+                io::ErrorKind::WouldBlock => continue,
+                _ => panic!("{}", format!("error while reading from pipe: {:?}", err)),
+            }
+            } else if let Ok(count) = res {
+                if count != payload.len() {
+                // If there is no data yet, just `continue` and try again.
+                    continue;
+                } else {
+                    let result: String = payload;
+                    println!("got data from client: {}", result);
```

Comment:
Don't use `println!`. Use `info!`, `warn!`, `debug!`, `error!` for meaningful logs.